From 0641cd9ac3eb142f239b70085bac459f3694eadd Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Thu, 21 Aug 2014 22:43:37 +1000
Subject: [PATCH 2/3] =?UTF-8?q?RingBuffer:=20always=20read=20directly=20from?=
 =?UTF-8?q?=20file=20if=20read=20ahead=20thread=20isn=E2=80=99t=20running?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

readInternalMode can only work if the read ahead thread is running
---
 mythtv/libs/libmythtv/ringbuffer.cpp |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/ringbuffer.cpp b/mythtv/libs/libmythtv/ringbuffer.cpp
index 0b855fa..3b23bf3 100644
--- a/mythtv/libs/libmythtv/ringbuffer.cpp
+++ b/mythtv/libs/libmythtv/ringbuffer.cpp
@@ -1426,8 +1426,7 @@ int RingBuffer::ReadPriv(void *buf, int count, bool peek)
         return -1;
     }
 
-    if (!readInternalMode &&
-        (request_pause || stopreads || !readaheadrunning || (ignorereadpos>=0)))
+    if (request_pause || stopreads || !readaheadrunning || (ignorereadpos>=0))
     {
         rwlock.unlock();
         rwlock.lockForWrite();
-- 
1.7.10.2

